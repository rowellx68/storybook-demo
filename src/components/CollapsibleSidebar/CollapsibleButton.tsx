/** @jsx jsx */
import React from 'react';
import { css, jsx } from '@emotion/core';

import { button } from './collapsible-shared-styles';

interface ButtonProps {
  expanded: boolean;
  icon: string;
  expandIcon?: string;
  label?: string;
  onClick: () => void;
}

const buttonStyles = css(button, {
  fontSize: '1.5rem',
});

const CollapsibleButton: React.FunctionComponent<ButtonProps> = ({
  onClick,
  expanded,
  icon,
  expandIcon,
  label,
}) => {
  return (
    <button css={buttonStyles} onClick={onClick}>
      <i
        className={expanded ? expandIcon ?? icon : icon}
        aria-hidden='true'></i>
      {expanded ? <span>{label}</span> : null}
    </button>
  );
};

export default CollapsibleButton;
