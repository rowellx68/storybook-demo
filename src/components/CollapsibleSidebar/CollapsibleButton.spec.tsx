import React from 'react';
import { mount } from 'enzyme';

import CollapsibleButton from 'components/CollapsibleSidebar/CollapsibleButton';

const icons = [
  'las la-fw la-angle-double-left',
  'las la-fw la-angle-double-right',
];

describe('CollapsibleButton', () => {
  it('renders the correct element - collapsed mark-up', () => {
    const component = mount(
      <CollapsibleButton
        expanded={false}
        onClick={() => null}
        icon={icons[0]}
        expandIcon={icons[1]}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('renders the correct element - expanded', () => {
    const component = mount(
      <CollapsibleButton
        expanded={true}
        onClick={() => null}
        icon={icons[0]}
        expandIcon={icons[1]}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
