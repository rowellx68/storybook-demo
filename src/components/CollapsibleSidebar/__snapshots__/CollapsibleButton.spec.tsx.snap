// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleButton renders the correct element - collapsed mark-up 1`] = `
.emotion-0 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: none;
  background: #f5f6fa;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4a5568;
  font-size: 1.5rem;
}

.emotion-0:hover {
  background: rgba(0,0,0,0.1);
}

@media screen and (min-width:640px) {
  .emotion-0 {
    margin-bottom: 0.5rem;
  }

  .emotion-0:last-of-type {
    margin-bottom: 0;
  }
}

.emotion-0.active {
  color: var(--primary);
  background: #ffffff;
}

.emotion-0 i {
  font-size: 1.5rem;
  opacity: 1;
}

.emotion-0 span {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1rem;
  padding: 0 0.75rem;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto', 'Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans', 'Helvetica Neue',sans-serif;
}

@media screen and (min-width:640px) {
  .emotion-0 span {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<CollapsibleButton
  expandIcon="las la-fw la-angle-double-right"
  expanded={false}
  icon="las la-fw la-angle-double-left"
  onClick={[Function]}
>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="las la-fw la-angle-double-left"
    />
  </button>
</CollapsibleButton>
`;

exports[`CollapsibleButton renders the correct element - expanded 1`] = `
.emotion-0 {
  padding: 0.5rem;
  border-radius: 0.25rem;
  border: none;
  background: #f5f6fa;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #4a5568;
  font-size: 1.5rem;
}

.emotion-0:hover {
  background: rgba(0,0,0,0.1);
}

@media screen and (min-width:640px) {
  .emotion-0 {
    margin-bottom: 0.5rem;
  }

  .emotion-0:last-of-type {
    margin-bottom: 0;
  }
}

.emotion-0.active {
  color: var(--primary);
  background: #ffffff;
}

.emotion-0 i {
  font-size: 1.5rem;
  opacity: 1;
}

.emotion-0 span {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1rem;
  padding: 0 0.75rem;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto', 'Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans', 'Helvetica Neue',sans-serif;
}

@media screen and (min-width:640px) {
  .emotion-0 span {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<CollapsibleButton
  expandIcon="las la-fw la-angle-double-right"
  expanded={true}
  icon="las la-fw la-angle-double-left"
  onClick={[Function]}
>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="las la-fw la-angle-double-right"
    />
    <span />
  </button>
</CollapsibleButton>
`;
